src/ResourceHelperTrait.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/*
+ * Copyright 2022 Google LLC
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace Google\ApiCore;
+
+use Google\ApiCore\ValidationException;
+
+trait ResourceHelperTrait
+{
+    private static $templateMap;
+
+    // Must be implemented by extendees to call loadTemplates.
+    private static function registerTemplates()
+    {
+        self::$templateMap = [];
+    }
+
+    private static function loadTemplates(string $configPath, string $serviceName)
```

Comment:
let's add return typehints for everything!
```suggestion
    private static function loadTemplates(string $configPath, string $serviceName): void
```

Reply:
Ok! Will do :)

Reply:
[Nullable typehints](https://www.php.net/manual/en/migration71.new-features.php) were added in PHP 7.1, so this will definitely be supported in the 2.x branch

Reply:
Ok, then I will add a TODO to add that for `getTemplate`.

Reply:
Actually refactored `loadTemplates` to not have a `return`, so let's see if that makes php 7 happy.

Reply:
Nope, need to remove it

Reply:
We can add it after we drop support for PHP 7.0 and below, but hopefully that'll happen before we roll this out anyway.